spark/v2.4/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkDataWrite.java

```diff
@@ -189,6 +192,56 @@ public void testAppend() throws IOException {
     Assert.assertEquals("Result rows should match", expected, actual);
   }
 
+  @Test
+  public void testConcurrentAppend() throws IOException {
```

Comment:
I don't think that we need tests for individual engines or versions. The catalog should be thoroughly tested and its behavior won't change when it is embedded in an engine.